runtime/mount/python_lib/lplots/h5/utils/align.py

```diff
@@ -15,16 +18,47 @@ class AlignmentMethod(Enum):
     stalign = "stalign"
 
 
-def align_image(
+async def capture_output(blocking_work: Callable[[], None], on_progress:
+                         Callable[[object], Awaitable[None]], stage: str,
+                         progress_msg_base: dict[str, any]) -> bool:
+    error_info: Exception | None = None
+    buf_out, buf_err = StringIO(), StringIO()
+    try:
+        with redirect_stdout(buf_out), redirect_stderr(buf_err):
+            blocking_work()
+            return False
+    except Exception as exc:
+        e = exc
+        error_info = traceback.format_exc()
+        return True
+    finally:
+        await on_progress(
+                {**progress_msg_base,
+                 "value": {
+                 "stage": stage,
+                 "stdout": buf_out.getvalue(),
+                 "stderr": buf_err.getvalue(),
+                 "error": error_info,
+                }})
+        if error_info is not None:
+            raise e
+
+
+async def align_image(
         scatter_data_key: str,
         new_scatter_data_key: str,
         points_I: list[list[float]],
         points_J: list[list[float]],
         alignment_method: AlignmentMethod,
         image_bytes: bytes | None,
         adata: ad.AnnData,
+        widget_session_key: str,
+        on_progress: Callable[[object], Awaitable[None]]
```

Comment:
This only really works as the callback name if you define it outside of there... otherwise this is just send